test-resources/geo/location-history-data-schema.json

```diff
@@ -8,6 +8,11 @@
     "version": {
       "const": 1
     },
```

Comment:
Shouldn't we change the version number to `2` since we're modifying the schema? 🤔

Reply:
Good question! I don't really have a clear understanding of how we're planning on using this version, to be honest. I'm not sure that I have any good opinions about it, either. Do you think that we should increment the version each time we change the schema?

Reply:
(I Googled "how to evolve a json schema" and it didn't help me very much!)

Reply:
We have an approach in `ably/ably-common` which may be helpful to refer to: https://github.com/ably/ably-common/tree/main/json-schemas#versioning

Reply:
I'd say that the version field should change with each new version of the file format 🤔 Otherwise we would have files that have the same version but contain different fields. I don't have any strong opinion on whether we should use the SemVer as suggested by Quintin or just use integer values, I'm happy with both options 😅

Reply:
@QuintinWillison I think I might not be understanding how the versioning in `ably-common` works. It says

> Version numbers follow semantic versioning.

That means, according to [Semantic Versioning](https://semver.org), that you "increment the PATCH version when you make backwards compatible bug fixes". My interpretation of "backwards compatible" in the context of a JSON schema is that a change from schema version $x$ to schema version $y$ is _backwards compatible_ if an object that conforms to schema version $x$ also conforms to schema $y$.

But, then I saw https://github.com/ably/ably-common/commit/2535f90ad28ab51f870cd87f015e144af8076bea, which seems to increment the PATCH version whilst removing a `required` property. This seems to violate the above rule. Have I misunderstood something here?

Reply:
> I'd say that the version field should change with each new version of the file format 🤔 Otherwise we would have files that have the same version but contain different fields. I don't have any strong opinion on whether we should use the SemVer as suggested by Quintin or just use integer values, I'm happy with both options 😅

Some thoughts on this. First of all, it's worth bearing in mind that this schema describes data that is going to be _archived_ – that is, saved and then potentially read a long time in the future. That means that as well as backwards compatibility (i.e. can an old version of the library read newer data), we also need to think about forwards compatibility (i.e. can a newer version of the library read older data).

I am thinking of the `version` field not as describing the version of the _schema_, but rather as data that just communicates information about backwards and forwards compatibility. So, I don't think we need to change the value that is saved in the `version` field each time we make _any_ change to the schema. That means that I don't think we need to adopt a versioning system as granular as semantic versioning.

Here's what I'm thinking. We stick with a single `version: int` field, which we only increment if we make a change that breaks backwards compatibility. Within a given version, we will never introduce a change that breaks forwards compatibility.

Reply:
I'm happy with that. I would suggest adding a readme to the folder containing these schemas that captures that definition.

I wasn't involved in the decision to adopt SemVer for schema versioning in `ably-common`, though I also would not have objected to that choice because we have been trying to conform to SemVer where we can, where it makes sense to. Perhaps @lmars might have a different perspective on whether AAT should 'break away' and keep it simpler by using integer versioning rather than SemVer. 🤔

Reply:
I'm also happy with that 😁 I'll leave this conversation unresolved in case you'd like to discuss it further. @lawrence-forooghian please feel free to resolve this conversation once you think it's no longer needed 👍

Reply:
After discussing, we're going to stick with semantic versioning. Will open another PR for the versioning decision and changes.

Reply:
I've opened #67 for the versioning, and changed the destination of this PR.